workflows/algolia/src/index.ts

```diff
@@ -0,0 +1,147 @@
+    title?: string;
+    description?: string;
+    slug?: string;
+    tags?: Array<string>;
+  };
+};
+
+const getAllFiles = function (
+  dirPath,
+  arrayOfFiles: Array<{ inputPath: string; outputPath: string }>
+) {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach(function (file) {
+    if (file === "LICENSE.MD") {
+      return;
+    }
+
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) {
+      arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles);
+    } else {
+      const inputPath = path.join(dirPath, "/", file);
+      const outputPath = `.\\output\\${path.join(
+        "./output/",
+        dirPath,
+        "/",
+        file
+      )}`;
+      arrayOfFiles.push({ inputPath, outputPath });
+    }
+  });
+
+  return arrayOfFiles;
+};
+
+export const checkObjectIds = () => {
+  const objectIds = [];
+
+  const checkObjectId = (matterData: DocsFrontMatter, type = "glossary") => {
+    if (!matterData.data.objectId) {
+      throw new Error(
+        `objectId missing on file ${type}/${matterData.data.slug}.`
+      );
+    }
+
+    if (objectIds.includes(matterData.data.objectId)) {
+      throw new Error(`objectId ${matterData.data.objectId} appeared twice.`);
+    }
+
+    objectIds.push(matterData.data.objectId);
+  };
+
+  const glossaryFiles = getAllFiles("../../resources/glossary", []);
+
+  for (const glossaryFileName of glossaryFiles) {
+    const file = fs.readFileSync(glossaryFileName.inputPath, "utf8");
+
+    const grayMatterData = matter(file) as unknown as DocsFrontMatter;
+
+    checkObjectId(grayMatterData);
+  }
+
+  const docsFiles = getAllFiles("../../resources/docs/simulation", []);
+
+  for (const docsFile of docsFiles) {
+    const file = fs.readFileSync(docsFile.inputPath, "utf8");
+
+    const grayMatterData = matter(file) as unknown as DocsFrontMatter;
+
+    checkObjectId(grayMatterData, "docs");
+  }
+};
+
+export const generateAlgoliaJson = () => {
+  const jsonData = [];
+  const objectIds = [];
+
+  const appendToJson = (matterData: DocsFrontMatter, type = "glossary") => {
+    if (!matterData.data.objectId) {
+      throw new Error(
+        `objectId missing on file ${type}/${matterData.data.slug}.`
+      );
+    }
+
+    if (objectIds.includes(matterData.data.objectId)) {
+      throw new Error(`objectId ${matterData.data.objectId} appeared twice.`);
+    }
+
+    objectIds.push(matterData.data.objectId);
+
+    const appendData = {
+      ...matterData.data,
+      objectId: undefined,
+      content: matterData.content,
+      objectID: matterData.data.objectId,
+      type,
+    };
+
+    jsonData.push(appendData);
+  };
+
+  const glossaryFiles = getAllFiles("../../resources/glossary", []);
+
+  for (const glossaryFileName of glossaryFiles) {
+    const file = fs.readFileSync(glossaryFileName.inputPath, "utf8");
+
+    const grayMatterData = matter(file) as unknown as DocsFrontMatter;
+
+    appendToJson(grayMatterData);
+  }
```

Comment:
I think you can combine this and the below to construct `jsonData` from a single `map`, since you're constructing a new array with the same logic.